### ISSUE TITLE
Nested entries

### ISSUE THREAD

**Issue by TheCymaera** [WICG/file-system-access#378]
It is common for IDEs and game-engine editors to specify resources using URIs or file paths. 

To access a deeply nested file, we need to call `getFileHandle` and `getDirectoryHandle` recursively. Each level is an asynchronous operation that needs to be run sequentially.

I propose that the `getFileHandle` and `getDirectoryHandle` methods be overloaded to accept an array/iterable argument.

```typescript
directory.getFileHandle(["path", "to", "file.txt"]);
```

Alternatively, file names should accept slashes as delimiters.
```typescript
directory.getFileHandle("path/to/file.txt");
```

The options parameter should include a `createRecursive` flag to complement the `create` flag.
```typescript
directory.getFileHandle(["path", "to", "file.txt"], { createRecursive: true });
// Creates:
// path/
// path/to/
// path/to/file.txt
```

**Comment by guillaumebrunerie** [comment]
You can easily implement it yourself, not sure if there would be an advantage in making the spec more complicated for that?

**Comment by TheCymaera** [comment]
Each level makes at least one file system call to check if the file exists. This takes anywhere between .2 to 1 milliseconds on the devices I've tried. The difference adds up quickly, especially when processing multiple files.

**Comment by SevenSystems** [comment]
This +1000. How is this not the default way to access files? File paths have been used since the 1970s, it is universally accepted and known. It's not just about performance. Many existing apps are designed to work with file paths (`foo/bar/test.txt`). Rewriting these to work with a directory from, say, `showDirectoryPicker` is needlessly complicated.

I mean, there's already `.resolve`, which does the reverse (although in a strange way returning an array instead of a path).

**Comment by SevenSystems** [comment]
@TheCymaera thanks for bringing this up. Maybe you should consider editing the title to be more precise, like "Add ability to use file paths to resolve a file" or something.

**Comment by guillaumebrunerie** [comment]
Would it actually improve performance? Surely the OS has to perform the same logic when resolving a full path?

**Comment by SevenSystems** [comment]
> Would it actually improve performance? Surely the OS has to perform the same logic when resolving a full path?

Yes, valid thinking. However, as this is such a common and performance-critical operation, the OS does this using extremely optimized low-level code, which is probably orders of magnitude faster than a manual loop using extremely abstracted asynchronous JavaScript APIs.
